### DISCUSSION TITLE
Cloning a repository on the same Windows network between two computers with LFS files

### DISCUSSION THREAD

**Discussion by junwago** [git-lfs/git-lfs#5692] [Q&A]
I have a problem cloning the repository between the same network of two computers in Windows 10. I execute the command
"git clone "file://DESKTOP-D3J67S2/Temas_Git/Repositorio"
and I get everything except the LFS contents. 
![ProblemaGit](https://github.com/git-lfs/git-lfs/assets/156134037/d601e5a7-4999-4dc5-b6c9-297d933ab34d)

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  Are you able to provide a bit more context regarding your configuration?  It looks like you have a network drive mounted and shared on two systems and are aiming to use a `file://` URL path to the Git repository you wish to clone, and the error message suggests the `/Temas_Git/Repositorio` directory is not found.

Could you provide the output of `git lfs env` when run within the cloned repository?

You might also try running with `GIT_TRACE=1` to see if there's a hint as to what's going wrong in the trace output.

**Reply by junwago** [reply]
When I run the command without lfs the clone downloads fine, the problem is when I use LFS.

Thanks, So this is the clone trace
-------------------------------------------------------------------------------------------------------------------

GIT_TRACE=1 git clone "file://DESKTOP-D3J67S2/Temas_Git/Repositorio"
08:31:27.006064 exec-cmd.c:244          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
08:31:27.021686 git.c:463               trace: built-in: git clone file://DESKTOP-D3J67S2/Temas_Git/Repositorio
Cloning into 'Repositorio'...
08:31:27.099437 run-command.c:659       trace: run_command: unset GIT_DIR; GIT_PROTOCOL=version=2 'git-upload-pack '\''//DESKTOP-D3J67S2/Temas_Git/Repositorio'\'''
08:31:27.294838 exec-cmd.c:244          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
08:31:27.334212 git.c:463               trace: built-in: git upload-pack //DESKTOP-D3J67S2/Temas_Git/Repositorio
08:31:27.474322 run-command.c:659       trace: run_command: git pack-objects --revs --thin --stdout --progress --delta-base-offset
08:31:27.474322 run-command.c:659       trace: run_command: git index-pack --stdin -v --fix-thin '--keep=fetch-pack 10056 on DESKTOP-D3J67S2' --check-self-contained-and-connected
remote: 08:31:27.489949 exec-cmd.c:244          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
08:31:27.505575 exec-cmd.c:244          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
08:31:27.536825 git.c:463               trace: built-in: git index-pack --stdin -v --fix-thin '--keep=fetch-pack 10056 on DESKTOP-D3J67S2' --check-self-contained-and-connected
remote: 08:31:27.536825 git.c:463               trace: built-in: git pack-objects --revs --thin --stdout --progress --delta-base-offset
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 4 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (4/4), done.
08:31:27.678186 run-command.c:659       trace: run_command: git rev-list --objects --stdin --not --all --quiet --alternate-refs '--progress=Checking connectivity'
08:31:27.693236 exec-cmd.c:244          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
08:31:27.708862 git.c:463               trace: built-in: git rev-list --objects --stdin --not --all --quiet --alternate-refs '--progress=Checking connectivity'
08:31:27.743123 run-command.c:659       trace: run_command: 'git-lfs filter-process'
08:31:27.844758 trace git-lfs: exec: uname
08:31:27.984175 trace git-lfs: exec: git 'version'
08:31:28.051773 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' '--git-dir' '--show-toplevel'
08:31:28.112280 trace git-lfs: exec: git 'rev-parse' '--is-bare-repository'
08:31:28.169820 trace git-lfs: exec: git 'config' '--includes' '--local' 'lfs.repositoryformatversion'
08:31:28.223652 trace git-lfs: exec: git 'config' '--includes' '--replace-all' 'lfs.repositoryformatversion' '0'
08:31:28.264954 trace git-lfs: exec: git 'config' '--includes' '-l'
08:31:28.325907 trace git-lfs: exec: git 'rev-parse' '--is-bare-repository'
08:31:28.418860 trace git-lfs: exec: git 'config' '--includes' '-l' '--blob' ':.lfsconfig'
08:31:28.510945 trace git-lfs: exec: git 'config' '--includes' '-l' '--blob' 'HEAD:.lfsconfig'
08:31:28.606587 trace git-lfs: Install hook: pre-push, force=false, path=C:\Users\Japo\Desktop\Clon\Repositorio\.git\hooks\pre-push
08:31:28.607587 trace git-lfs: Install hook: post-checkout, force=false, path=C:\Users\Japo\Desktop\Clon\Repositorio\.git\hooks\post-checkout
08:31:28.608590 trace git-lfs: Install hook: post-commit, force=false, path=C:\Users\Japo\Desktop\Clon\Repositorio\.git\hooks\post-commit
08:31:28.610589 trace git-lfs: Install hook: post-merge, force=false, path=C:\Users\Japo\Desktop\Clon\Repositorio\.git\hooks\post-merge
08:31:28.614753 trace git-lfs: Initialize filter-process
08:31:28.617751 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
08:31:28.699870 trace git-lfs: tq: running as batched queue, batch size of 100
08:31:28.716204 trace git-lfs: filepathfilter: accepting "Doc_01.txt"
08:31:28.717475 trace git-lfs: tq: sending batch of size 1
08:31:28.717783 trace git-lfs: tq: starting transfer adapter "lfs-standalone-file"
08:31:28.729725 trace git-lfs: exec: sh '-c' 'git-lfs standalone-file'
08:31:29.208389 trace git-lfs: xfer[git-lfs]: error creating handler: chdir /Temas_Git/Repositorio: El sistema no puede encontrar la ruta especificada.
08:31:29.209486 trace git-lfs: tq: running as batched queue, batch size of 100
08:31:29.209486 trace git-lfs: filepathfilter: accepting "Doc_01.txt"
Downloading Doc_01.txt (6 B)
08:31:29.210301 trace git-lfs: tq: running as batched queue, batch size of 100
08:31:29.210813 trace git-lfs: tq: sending batch of size 1
08:31:29.210813 trace git-lfs: tq: starting transfer adapter "lfs-standalone-file"
08:31:29.218181 trace git-lfs: exec: sh '-c' 'git-lfs standalone-file'
08:31:29.723444 trace git-lfs: xfer[git-lfs]: error creating handler: chdir /Temas_Git/Repositorio: El sistema no puede encontrar la ruta especificada.
Error downloading object: Doc_01.txt (e470fa0): Smudge error: Error downloading Doc_01.txt (e470fa0d001e50b3ec3088022462a94ea7c87dd80106411b7d120f90b379e977): error initializing custom adapter "lfs-standalone-file" worker 0: [0] error creating handler: chdir /Temas_Git/Repositorio: El sistema no puede encontrar la ruta especificada.

Errors logged to 'C:\Users\Japo\Desktop\Clon\Repositorio\.git\lfs\logs\20240405T083129.7239533.log'.
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: Doc_01.txt: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/


This is the git lfs env trace
-------------------------------------------------------------------------------------------------------------------
$ GIT_TRACE=1 git lfs env
08:35:07.683667 exec-cmd.c:244          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
08:35:07.699293 git.c:749               trace: exec: git-lfs env
08:35:07.699293 run-command.c:659       trace: run_command: git-lfs env
08:35:07.743152 trace git-lfs: exec: git 'version'
git-lfs/3.4.0 (GitHub; windows amd64; go 1.20.6; git d06d6e9e)
git version 2.43.0.windows.1
08:35:07.969440 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
08:35:08.022630 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' '--git-dir' '--show-toplevel'
08:35:08.080914 trace git-lfs: exec: uname
08:35:08.158600 trace git-lfs: exec: git 'config' '--includes' '-l'
08:35:08.212578 trace git-lfs: exec: git 'rev-parse' '--is-bare-repository'
08:35:08.267333 trace git-lfs: exec: git 'config' '--includes' '-l' '--blob' ':.lfsconfig'
08:35:08.324801 trace git-lfs: exec: git 'config' '--includes' '-l' '--blob' 'HEAD:.lfsconfig'
Endpoint=file://DESKTOP-D3J67S2/Temas_Git/Repositorio (auth=none)
LocalWorkingDir=C:\Users\Japo\Desktop\Clon\Repositorio
LocalGitDir=C:\Users\Japo\Desktop\Clon\Repositorio\.git
LocalGitStorageDir=C:\Users\Japo\Desktop\Clon\Repositorio\.git
LocalMediaDir=C:\Users\Japo\Desktop\Clon\Repositorio\.git\lfs\objects
LocalReferenceDirs=
TempDir=C:\Users\Japo\Desktop\Clon\Repositorio\.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=C:\Users\Japo\Desktop\Clon\Repositorio\.git\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_TRACE=1
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
08:35:08.380604 trace git-lfs: filepathfilter: creating pattern ".git" of type gitignore
08:35:08.380604 trace git-lfs: filepathfilter: creating pattern "**/.git" of type gitignore
08:35:08.380604 trace git-lfs: filepathfilter: accepting "tmp"

**Comment by chrisd8088** [comment]
Hmm ... it looks like the `chdir` is being [attempted](https://github.com/git-lfs/git-lfs/blob/19a702e780c0ad93f0e02d9c96adbf5fd7da70bb/lfshttp/standalone/standalone.go#L122) in the `git-lfs-standalone-file` transfer adapter, and for whatever reason, it's had the leading `DESKTOP-D3J67S2` volume name (I assume that's what that is) removed.  I'll try to see if there's a way to replicate this on a Windows VM.

**Reply by chrisd8088** [reply]
I think one issue here, at least, is that the URL provided starts with `file://` instead of `file:///`.  As a result, the first component is treated as a hostname, not a path component.  The `git-lfs-standalone-file` transfer adapter [parses](https://github.com/git-lfs/git-lfs/blob/19a702e780c0ad93f0e02d9c96adbf5fd7da70bb/lfshttp/standalone/standalone.go#L175) the URL, and then only [references](https://github.com/git-lfs/git-lfs/blob/19a702e780c0ad93f0e02d9c96adbf5fd7da70bb/lfshttp/standalone/standalone.go#L183) the `url.Path` element, so the `url.Host` element is never used.

Arguably, what the Git LFS standalone file adapter should be doing in this case, at least on Windows, is to convert a `file://` URL into a UNC path as described in this [blog post](https://learn.microsoft.com/en-us/archive/blogs/ie/file-uris-in-windows), as mentioned in golang/go#32456, so that `file://DESKTOP-D3J67S2/Temas_Git/Repositorio` would become `\\DESKTOP-D3J67S2\Temas_Git\Repositorio`.  To do this, we could borrow Go's [implementation](https://github.com/golang/go/blob/ecfddf4841ceaa7fd253aae0b6d9d0d59625674b/src/cmd/go/internal/web/url_windows.go#L13-L43) from their command-line tools.

I'll write up an issue describing the problem.  Thanks for the report!

**Reply by chrisd8088** [reply]
I've opened #5703 with the details from this discussion.
